scripts/config

```diff
@@ -9,115 +9,74 @@
-then
-    # If fpm_footprint is an integer, that's a numeric value for the footprint
-    old_free_footprint=$fpm_footprint
-else
-    old_free_footprint=0
-fi
-free_footprint="${YNH_CONFIG_MAIN_PHP_FPM_CONFIG_FREE_FOOTPRINT:-$old_free_footprint}"
-
-# Usage for PHP-FPM
-old_fpm_usage="$(ynh_app_setting_get --app=$app --key=fpm_usage)"
-fpm_usage="${YNH_CONFIG_MAIN_PHP_FPM_CONFIG_USAGE:-$old_fpm_usage}"
-
 #=================================================
-# SHOW_CONFIG FUNCTION FOR 'SHOW' COMMAND
+# SPECIFIC VALIDATORS FOR TOML SHORT KEYS
 #=================================================
 
-show_config() {
-	# here you are supposed to read some config file/database/other then print the values
-	# ynh_return "YNH_CONFIG_${PANEL_ID}_${SECTION_ID}_${OPTION_ID}=value"
-	ynh_return "YNH_CONFIG_MAIN_SFTP_SFTP=$with_sftp"
-# 	ynh_return "YNH_CONFIG_MAIN_SFTP_PASSWORD=$password"
+#=================================================
+# SPECIFIC SETTERS FOR TOML SHORT KEYS
+#=================================================
 
-	ynh_return "YNH_CONFIG_MAIN_PHP_FPM_CONFIG_FOOTPRINT=$fpm_footprint"
-	ynh_return "YNH_CONFIG_MAIN_PHP_FPM_CONFIG_FREE_FOOTPRINT=$free_footprint"
-	ynh_return "YNH_CONFIG_MAIN_PHP_FPM_CONFIG_USAGE=$fpm_usage"
+set__password() {
+    if [ "$password" == "" ]
+    then
+        ynh_app_setting_set --app=$app --key=password --value="$password"
+    fi
 }
 
 #=================================================
-# MODIFY THE CONFIGURATION
+# GENERIC FINALIZATION
 #=================================================
 
-apply_config() {
-    #=================================================
-    # REMOVE OR ADD SFTP ACCESS
-    #=================================================
+ynh_app_config_validate() {
+    _ynh_app_config_validate
 
-    if [ "$with_sftp" != "$old_with_sftp" ]
+    if [ "${changed[with_sftp]}" == "true" ] && [ $with_sftp -eq 1 ] && [ "$password" == "" ]
     then
-        yunohost app action run $app sftp --args with_sftp=$with_sftp
+        ynh_die --message="You need to set a password to enable SFTP"
+    fi
+}
 
-        # Change the password only if none was already set for the user
-        if [ $is_password_exist -eq 0 ] && [ $with_sftp -eq 1 ]
-        then
-            # Add the password to the user
-            chpasswd <<< "${app}:${password}"
-            ynh_app_setting_set --app=$app --key=password --value="$password"
-        fi
+ynh_app_config_apply() {
+    _ynh_app_config_apply
+
+
+    if [ "${changed[phpversion]}" == "true" ] && [ "$phpversion" == "none" ]
+    then
+        # FIXME: This miserably fails because the setting phpversion get his 
+        # new value before ynh_app_config_apply gets called, so 
+        # ynh_remove_fpm_config try to remove the new not-yet-existent version.
+        ynh_remove_fpm_config
+    elif [ "${changed[phpversion]}" == "true" ]
+    then
+        ynh_add_fpm_config --usage=low --footprint=low --phpversion=$phpversion
```

Comment:
@ericgaspar I guess your issue lies at this line actually. Untested code, still lazily cheating:

```suggestion
        new_phpversion=$phpversion
        phpversion=$(ynh_app_setting_get --app=$app --key=phpversion)
        ynh_remove_fpm_config
        phpversion=$new_phpversion
        ynh_add_fpm_config --usage=low --footprint=low --phpversion=$phpversion
```